### DISCUSSION TITLE
ARABIC-INDIC DIGITs don't appear after U+6DD in hb-view with Amiri typeface

### DISCUSSION THREAD

**Discussion by munzirtaha** [harfbuzz/harfbuzz#4607] [Q&A] [answered]
`$ hb-view /usr/share/fonts/TTF/Amiri-Regular.ttf -u '6DD,661'`

![hb-amiri](https://github.com/harfbuzz/harfbuzz/assets/2936425/04fa1d51-c47c-4623-97be-7372ed7042df)

I don't see the problem in LibreOffice

**Comment by khaledhosny** [comment, marked as answer]
You need `--direction=ltr` (since hb-view will misautodetect it as `rtl`). Also the digits are there but mispositioned outside U+6DD, if you add big enough margins you will see them.

**Comment by munzirtaha** [comment]
Thanks, yes I can see it now with a 180 margin, but shouldn't subtending work in both directions? In LibreOffice it works with `Right-To-Left` direction.

Why it works for other fonts like `NotoSansArabic` but not `Amiri`?

**Reply by khaledhosny** [reply]
Arabic digits and subtending marks are strong left to right characters, so they always get LTR direction after applying bidirectional text algorithm even in RTL text (hb-view/shape tools don’t apply bidi algorithm, so for cases like this you need to set the direction explicitly). Amiri uses pairwise kerning to position the digits, Noto Sans Arabic using mark positioning, so setting the wrong direction can have different effects on both fonts.

**Reply by munzirtaha** [reply]
Ok, thanks for the clarification.
